src/batch-client/index.ts

```diff
@@ -551,6 +551,43 @@ export class ZimbraBatchClient {
 			}
 		}).then(res => normalize(Folder)(res.folder[0].folder));
 
+	public uploadMessage = (messageString: string) => {
+		const contentDisposition = 'attachment';
+		const filename = 'message.eml';
+		const contentType = 'message/rfc822';
+
+		return fetch(`${this.origin}/service/upload?fmt=extended,raw`, {
+			method: 'POST',
+			body: messageString,
+			headers: {
+				'Content-Disposition': `${contentDisposition}; filename="${filename}"`,
+				'Content-Type': contentType
+			},
+			credentials: 'include'
+		}).then(response => {
+			if (response.ok) {
+				return response.text().then(result => {
+					if (!result) {
+						return null;
+					}
+
+					// To parser server response like => 200,'null',[{"aid":"d93a252a-603e-4675-9e39-95cebe5a9332:b39a4b7c-9232-4228-9269-aa375bc1df67","ct":"message/rfc822","filename":"message.eml","s":7872}]
```

Comment:
if we are only using attachmentId then can we use raw format instead of extended,raw